src/Control/Monad/Lift/Linear.hs

```diff
@@ -0,0 +1,39 @@
+-- | A class to lift non-linear monads into linear monads.
+--
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE NoImplicitPrelude #-}
+{-# LANGUAGE TypeFamilies #-}
+module Control.Monad.Lift.Linear where
+
+import qualified Control.Monad as NonLinear
+import qualified Control.Monad.Linear as Linear
+import Prelude.Linear
+import qualified System.IO.Linear as Linear
+
+-- | Lift a non-linear monad into a linear monad.
+--
+-- Laws:
+--
+-- > lift . return = return
+--
+-- > liftU . return = return . Unrestricted
+--
+-- > lift (m >>= f) = liftU m >>= \(Unrestricted a) -> lift (f a)
+--
+-- > liftU (m >>= f) = liftU m >>= \(Unrestricted a) -> liftU (f a)
+--
+class (Linear.Monad m, NonLinear.Monad (LiftedM m)) => MonadLift m where
+
+  -- | The type of the monad whose computations are lifted
+  type LiftedM m :: * -> *
+
+  -- | Lift a nonlinear monadic computation into a linear monad
+  lift :: LiftedM m a -> m a
```

Comment:
`lift` admits an implementation in terms of `liftU`, we ought to provide it as a default.